spark-ui/src/reducers/SqlReducer.ts

```diff
@@ -485,3 +487,75 @@ function calcBroadcastExchangeDuration(
   }
   return undefined;
 }
+
+function updateNodeMetrics(
+  node: EnrichedSqlNode,
+  metrics: EnrichedSqlMetric[],
+  graph: Graph,
+  allNodes: EnrichedSqlNode[],
+): EnrichedSqlMetric[] {
+
+  const updatedMetrics = calcNodeMetrics(node.type, metrics);
+  addFilterRatioMetric(node, updatedMetrics, graph, allNodes);
```

Comment:
you return updatedMetrics and does nothing with it.
We prefer to be functional, so add the new metric and with [..., newMetric"] and return this from the method

Reply:
changed it